airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -416,12 +417,16 @@ async def _run(self, connector_under_test_container: Container) -> StepResult:
         container = container.with_(hacks.never_fail_exec(self.regression_tests_command()))
         regression_tests_artifacts_dir = str(self.regression_tests_artifacts_dir)
         path_to_report = f"{regression_tests_artifacts_dir}/session_{self.run_id}/report.html"
-        await container.file(path_to_report).export(path_to_report)
 
         exit_code, stdout, stderr = await get_exec_result(container)
 
-        with open(path_to_report, "r") as fp:
-            regression_test_report = fp.read()
+        try:
+            await container.file(path_to_report).export(path_to_report)
+            with open(path_to_report, "r") as fp:
+                regression_test_report = fp.read()
+        except Exception as exc:
+            main_logger.exception(f"Could not export or read report due to exception. traceback={traceback.format_exc()}", exc_info=exc)
+            regression_test_report = None
```

Comment:
Just summarizing my understanding of the flow.
* `get_exec_result` actually executes the command in the container
* If there's an `ExecError` during the execution of this command we get stdout and stderr from the exception object
* If an `ExecError` was thrown (and handled) executing `container.file()` will likely fail as we're trying to get a file that does not exist.

Instead of excepting a broad exception I'd be in favor of being more specific:
```python
if "report.html" not in await container.directory(f"{regression_tests_artifacts_dir}/session_{self.run_id}").entries():
   main_logger.exception("The report file was not generated, an unhandled error likely happened during regression test execution, please check the step stderr and stdout for more details")
else:
  await container.file(path_to_report).export(path_to_report)
  with open(path_to_report, "r") as fp:
    regression_test_report = fp.read()
```

Reply:
I like the suggestion, updated.